com/williamfiset/datastructures/binarysearchtree/SplayTree.java

```diff
@@ -0,0 +1,274 @@
+import com.williamfiset.datastructures.utils.TreePrinter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+class BinaryTree<T extends Comparable<T>> implements TreePrinter.PrintableNode {
+    private BinaryTree<T> left, right;
+    private T data;
+
+    public BinaryTree(T data) {
+        this.data = data;
+    }
+
+    @Override
+    public BinaryTree<T> getLeft() {
+        return left;
+    }
+
+    @Override
+    public BinaryTree<T> getRight() {
+        return right;
+    }
+
+    @Override
+    public String getText() {
+        return data.toString();
+    }
+    public void setRight(BinaryTree<T> right) {
+        this.right = right;
+        //   right.parent = this;
+    }
+    public void setLeft(BinaryTree<T> left) {
```

Comment:
We never want to allow null data into the tree. Please check this and throw an error. Same for setRight, setData and the constructor.